server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java

```diff
@@ -490,6 +490,16 @@ private ShardCollectorProvider getCollectorProviderSafe(ShardId shardId) {
                     ShardCollectorProvider shardCollectorProvider = getCollectorProviderSafe(shardId);
                     shardRowContexts.add(shardCollectorProvider.shardRowContext());
                 } catch (ShardNotFoundException | IllegalIndexShardStateException e) {
```

Comment:
Could split this into two `catch` clauses to avoid a redundant `e instanceof ShardNotFoundException` further below